src/openms/source/ANALYSIS/XLMS/OpenProXLUtils.cpp

```diff
@@ -0,0 +1,2204 @@
+      }
+
+      p_cumul += coeff * pow(p,  j) * pow((1-p), (n-j));
+    }
+
+    // match-odds score becomes INFINITY for p_cumul >= 1, p_cumul might reach 1 because of insufficient precision, solved by using largest value smaller than 1
+    if (p_cumul >= 1.0)
+    {
+      p_cumul = nexttoward(1.0, 0.0);
+    }
+
+    return p_cumul;
+  }
+
+  // match odds score, spectra must be sorted by position
+  double OpenProXLUtils::match_odds_score(const PeakSpectrum& theoretical_spec,  const std::vector< std::pair< Size, Size > >& matched_spec, double fragment_mass_tolerance, bool fragment_mass_tolerance_unit_ppm, bool is_xlink_spectrum, Size n_charges)
+  {
+    // TODO Theoretical spectra for cross-links contain 1. and 2. isotopic peak, is mostly one of them matched making theo_size = 2 * matched_size in the best case?
+    // how does that skew the statistics?
+    // should we use theo_size / 2 for cross-links?
+    // or n_charges * 2?
+    Size matched_size = matched_spec.size();
+    Size theo_size = theoretical_spec.size();
+
+    if (matched_size < 1 || theo_size < 1)
+    {
+      return 0;
+    }
+
+    double range = theoretical_spec[theo_size-1].getMZ() -  theoretical_spec[0].getMZ();
+
+    // Compute fragment tolerance for the middle of the range / mean of MZ values, if ppm
+    // TODO mean should be used, so sum over MZs and devide by number, if ppm
+    double mean = 0.0;
+    for (Size i = 0; i < theo_size; ++i)
+    {
+      mean += theoretical_spec[i].getMZ();
+    }
+    mean = mean / theo_size;
+    double tolerance_Th = fragment_mass_tolerance_unit_ppm ? mean * 1e-6 * fragment_mass_tolerance : fragment_mass_tolerance;
+
+    // A priori probability of a random match given info about the theoretical spectrum
+    //    double a_priori_p = a_priori_probability(tolerance_Th, theo_size, range, 3);
+    double a_priori_p = 0;
+
+    if (is_xlink_spectrum)
+    {
+      a_priori_p = (1 - ( pow( (1 - 2 * tolerance_Th / (0.5 * range)),  (theo_size / n_charges))));
+    }
+    else
+    {
+      a_priori_p = (1 - ( pow( (1 - 2 * tolerance_Th / (0.5 * range)),  (theo_size))));
+    }
+
+    double match_odds = 0;
+    match_odds = -log(1 - cumulativeBinomial(theo_size, matched_size, a_priori_p) + 1e-5);
+    //recompile this shit fo shure
+
+//     cout << "TEST a_priori_prob: " << a_priori_p << " | tolerance: " << tolerance_Th << " | theo_size: " << theo_size << " | matched_size: " << matched_size << " | cumul_binom: " << cumulativeBinomial(theo_size, matched_size, a_priori_p)
+//              << " | match_odds: " << match_odds << endl;
+
+    // score lower than 0 does not make sense, but can happen if cumBinom = 0, -log( 1 + 1e5 ) < 0
+    if (match_odds >= 0.0)
+    {
+      return match_odds;
+    }
+    else
+    {
+      return 0;
+    }
+  }
+
+//   // Cross-correlation, with shifting the second spectrum from -maxshift to +maxshift of tolerance bins (Tolerance in Da, a constant binsize)
```

Comment:
please remove commented-out code : is this ever going to be useful again or why is this here?